src/transforms/filter.rs

```diff
@@ -0,0 +1,55 @@
+use super::Transform;
+use crate::{
+    conditions::{AnyCondition, Condition},
+    event::Event,
+    topology::config::{DataType, TransformConfig, TransformContext, TransformDescription},
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+struct FilterConfig {
+    condition: AnyCondition,
+}
+
+inventory::submit! {
+    TransformDescription::new_without_default::<FilterConfig>("filter")
+}
+
+#[typetag::serde(name = "filter")]
+impl TransformConfig for FilterConfig {
+    fn build(&self, _cx: TransformContext) -> crate::Result<Box<dyn Transform>> {
+        Ok(Box::new(Filter::new(self.condition.build()?)))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Any
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Any
```

Comment:
This is ultimately what we want but (from my current understanding) we can't make use of `Any` until https://github.com/timberio/vector/issues/421 is finished.

Reply:
Ok, I will revert this to `DataType::Log`